libsqlite3-sys/sqlite3/sqlite3.c

```diff
@@ -230515,3 +230515,19 @@ SQLITE_API int sqlite3_stmt_init(
 /* Return the source-id for this library */
 SQLITE_API const char *sqlite3_sourceid(void){ return SQLITE_SOURCE_ID; }
 /************************** End of sqlite3.c ******************************/
+
+#if !defined(__clang__)
+// The following functions are taken from https://www.openwall.com/lists/musl/2015/06/17/1
+// and fill in for the missing libc support functions generatd by `-D_FORTIFY_SOURCE=2`.
+void *__memcpy_chk(void *restrict dest, const void *restrict src, size_t n, size_t destlen)
+{
+  assert(n <= destlen);
+	return memcpy(dest, src, n);
+}
+
+void *__memset_chk(void *dest, int c, size_t n, size_t destlen)
+{
+  assert(n <= destlen);
+	return memset(dest, c, n);
+}
+#endif
```

Comment:
maybe these should be echoed in by the build script to make it easer to update the sqlite amalgamation?

Reply:
We shouldn't modify the sqlite3.c and should just add another .c file for this, I think.

We could even expose them as symbols from rust honestly...